tests/unit/test-util/test.artifacts.js

```diff
@@ -55,4 +56,38 @@ describe('prepareArtifactsDir', () => {
     }
   ));
 
+  it('throws error when lacking writing permissions', () => withTempDir(
+    (tmpDir) => {
+      const tmpPath = path.join(tmpDir.path(), 'build');
+      return fs.mkdir(tmpPath, '0622').then(() => {
+        const artifactsDir = path.join(tmpPath, 'artifacts');
+        return prepareArtifactsDir(artifactsDir)
+          .then(makeSureItFails())
+          .catch(onlyInstancesOf(UsageError, (error) => {
+            assert.match(error.message, /lack permissions/);
+          }));
+      });
+    }
+  ));
+
+  it('throws error when creating writing permissions', () => withTempDir(
+    (tmpDir) => {
+      class ErrorWithCode extends Error {
+        code: string;
+        constructor() {
+          super('some error');
+          this.code = 'EACCES';
+        }
+      }
+
+      const tmpPath = path.join(tmpDir.path(), 'build');
+      fs.mkdir = sinon.spy(() => Promise.reject(new ErrorWithCode()));
```

Comment:
@rpl It is indeed not the safest way, thanks for pointing it out. I wonder if it make sense to rewrite the first test accordingly?